coding-guides/raw-events.md

```diff
@@ -40,30 +40,31 @@ Let's add a `raw` event listener. Usually the given items are called a packet, s
 
 ```javascript
 client.on('raw', packet => {
-    // We don't want this to run on unrelated packets
-    if (!['MESSAGE_REACTION_ADD', 'MESSAGE_REACTION_REMOVE'].includes(packet.t)) return;
-    // Grab the channel to check the message from
-    const channel = client.channels.get(packet.d.channel_id);
-    // There's no need to emit if the message is cached, because the event will fire anyway for that
-    if (channel.messages.has(packet.d.message_id)) return;
-    // Since we have confirmed the message is not cached, let's fetch it
-    channel.fetchMessage(packet.d.message_id).then(message => {
-        // Emojis can have identifiers of name:id format, so we have to account for that case as well
-        const emoji = packet.d.emoji.id ? `${packet.d.emoji.name}:${packet.d.emoji.id}` : packet.d.emoji.name;
-        // This gives us the reaction we need to emit the event properly, in top of the message object
-        const reaction = message.reactions.get(emoji);
-        // Adds the currently reacting user to the reaction's users collection.
-        if (reaction) reaction.users.set(packet.d.user_id, client.users.get(packet.d.user_id));
-        // Check which type of event it is before emitting
-        if (packet.t === 'MESSAGE_REACTION_ADD') {
-            client.emit('messageReactionAdd', reaction, client.users.get(packet.d.user_id));
-        }
-        if (packet.t === 'MESSAGE_REACTION_REMOVE') {
-            client.emit('messageReactionRemove', reaction, client.users.get(packet.d.user_id));
-        }
-    });
-});
+   // We don't want this to run on unrelated packets
+   if (!['MESSAGE_REACTION_ADD', 'MESSAGE_REACTION_REMOVE'].includes(packet.t)) { return; }
+   // Grab the channel to check the message from
+   const channel = client.channels.resolve(packet.d.channel_id);
+   // There's no need to emit if the message is cached, because the event will fire anyway for that
+   if (channel.messages.cache.get(packet.d.message_id)) { return; }
+   // Since we have confirmed the message is not cached, let's fetch it
+   channel.messages.fetch(packet.d.message_id).then(function (message) {
+      // Emojis can have identifiers of name:id format, so we have to account for that case as well
+      const emoji = packet.d.emoji.id ? "" + packet.d.emoji.id : packet.d.emoji.name;
```

Comment:
This I do not agree with at all, why change it from a perfectly valid template literal to `"" + `, it makes no sense and makes the code inconsistent with the rest of the guide where template literals are used.

Reply:
because it didnt work in v12 anymore ... this part was kinda black magic and wasnt in the discord.js documentation ... so I just went off what was in other objects iirc ... its been like 2 months since I touched this though so its a little foggy

Reply:
Using string concatenation is the problem, not the order. There is no way you need to do `"" + packet.d.emoji.id` for this to be valid, because ID is already a string. Are you 100% certain that the identifiers have changed from name:id to just id? Because I have to doubt this very strongly, that would be a huge API breaking change.

Reply:
"that would be a huge API breaking change"
one among many ...

again, this part is not well documented, or at least not anywhere I could find ... i based it off the Object i was receiving with out this code.